board.go

```diff
@@ -39,42 +41,39 @@ func (b *board) countNeighbours(x, y int) int {
 }
 
 func (b *board) nextGen() {
-	state := b.computeNextGen()
+	b.computeNextGen()
 	b.generation++
-
-	for y := 0; y < b.height; y++ {
-		for x := 0; x < b.width; x++ {
-			b.cells[y][x] = state[y][x]
-		}
-	}
+	b.nextGenCells, b.cells = b.cells, b.nextGenCells
 }
 
-func (b *board) computeNextGen() [][]bool {
-	state := make([][]bool, b.height)
-
-	for y := 0; y < b.height; y++ {
-		state[y] = make([]bool, b.width)
-
-		for x := 0; x < b.width; x++ {
-			n := b.countNeighbours(x, y)
-
-			if b.cells[y][x] {
-				state[y][x] = n == 2 || n == 3
-			} else {
-				state[y][x] = n == 3
-			}
+func (b *board) computeNextGen() {
+	width, height := b.width, b.height
+	for y := 0; y < height; y++ {
+		for x := 0; x < width; x++ {
+			go func(x,y int) {
+				n := b.countNeighbours(x, y)
+				if b.cells[y][x] {
+					b.nextGenCells[y][x] = n == 2 || n == 3
+				} else {
+					b.nextGenCells[y][x] = n == 3
+				}
+				b.ch<-true
```

Comment:
I think this is local to the method, so does it need to be in the game state?
Also I wonder, does it not fit the `WorkGroup` model better?